### ISSUE TITLE
Style/MapToHash suggestion breaks code (again)

### ISSUE THREAD

**Issue by forthrin** [rubocop/rubocop#14780]
## Steps to reproduce the problem

```ruby
array = [[:a, 1], [:b, 2]]
array.map { [it.last, it.first] }.to_h.then { pp it } # Style/MapToHash
array.to_h { [it.last, it.first] }.then { pp it }

hash = array.to_h
hash.map { [it.last, it.first] }.to_h.then { pp it } # Style/MapToHash
hash.to_h { [it.last, it.first] }.then { pp it }
```

## Expected behavior

Don't complain? But also confounded about the element of possible programmer stupidity / language abuse. The interchangeability between hashes and arrays in Ruby sometimes breaks, assuming by careful design.

## Actual behavior

```
{1 => :a, 2 => :b}
{1 => :a, 2 => :b}
{1 => :a, 2 => :b}
undefined method 'last' for an instance of Symbol (NoMethodError)
hash.to_h { [it.last, it.first] }.then { pp it }
               ^^^^^
```

## RuboCop version

```
System Version: macOS 26.1 (25B78)
Model Identifier: MacBookAir10,1
ruby 4.0.0 (2025-12-25 revision 553f1675f3) +PRISM [arm64-darwin25]
rubocop     2026-01-04 99fa0fdd0
```

See also #14416

**Comment by koic** [comment]
`Style/MapToHash` is already marked as unsafe. If you expect safe autocorrect, please use `-a` instead of `-A`.
https://docs.rubocop.org/rubocop/cops_style.html#stylemaptohash

**Comment by forthrin** [comment]
Didn't expect autocorrect, but I found out, at least in my use cafe, that the proper solution was to do as follows. Is this a solution that is detectable and may work as a more reliable suggestion for replacement?

```
hash.map      { ... }.to_h # Warning
hash.to_h     { ... }      # Error
hash.to_a.map { ... }.to_h # Success
```
